crypto/ct/ct_sct.c

```diff
@@ -372,10 +377,28 @@ int SCT_validate(SCT *sct, const CT_POLICY_EVAL_CTX *ctx)
             goto err;
     }
 
+    /*
+     * XXX: Potential for optimization.  This repeats some idempotent heavy
+     * lifting on the certificate for each candidate SCT, and appears to not
+     * use any information in the SCT itself, only the certificate is
+     * processed.  So it may make more sense to to do this just once, perhaps
+     * associated with the shared (by all SCTs) policy eval ctx.
+     *
+     * XXX: Failure here is global (SCT independent) and represents either an issue
+     * with the certificate (e.g. duplicate extensions) or an out of memory
+     * condition.  When the certificate is incompatible with CT, we really
```

Comment:
Agreed.  Wrote comment before deciding to update the code.  See fixup commit (soon).